_posts/2016-04-25-watson-and-rails.md

```diff
@@ -0,0 +1,213 @@
+end
+```
+
+The `WatsonApi` wrapper class would take care of sending the request to the
+API.
+
+I used the [faraday](https://github.com/lostisland/faraday) gem to format the
+HTTP request. The `WatsonApi` model contained the following:
+
+```ruby
+app/models/watson_api.rb
+
+require 'json'
+require 'open-uri' #For opening the file when not stored locally
+require 'faraday'
+
+class WatsonApi
+  attr_reader :data
+
+  # Methods called on the Assignment show page when you call WatsonApi.new
+
+  def initialize(doc) #doc argument is the @assignment.file
+    if Rails.env.production?
+      # In production, the file is a url from an S3 bucket
+      @doc = doc.url
+    else
+      # In development, it's a local storage path
+      @doc = doc.path
+    end
+    # This calls the get_doc method whenever we make a WatsonApi.new request
+    # from the show page and sets the response data equal to the @data variable
+    @data = get_doc
+  end
+
+  def questions
+    # This iterates through the returned JSON and pulls out the desired data,
+    # in this case the questions from the Assignment.
+    @data.values[3].map { |item| [item["title"], item["content"][0]["text"]] }
+  end
+
+  private
+
+  def get_doc
+
+    # Connecting to the API with authorizations
+
+    base_url = "https://gateway.watsonplatform.net"
+    version = "2015-12-15"
+    path = "/document-conversion/api/v1/convert_document?version=#{version}"
+    basic_auth_username = ENV["WATSON_USERNAME"]
+    basic_auth_password = ENV["WATSON_PASSWORD"]
+    # Env variables in your .env file, given to you by signing up with Watson
+
+    conn = Faraday.new(url: base_url) do |faraday|
+      faraday.request :multipart #multipart indicates that a file is being uploaded
+      faraday.request :url_encoded
+      faraday.adapter :net_http
+    end
+
+    conn.basic_auth(basic_auth_username, basic_auth_password)
+
+    # Setting the payload configuration and adding your file to the request
+
+    payload = {}
+    if Rails.env.production?
+      payload = {
+        config: Faraday::UploadIO.new("#{::Rails.root}/lib/WatsonApi/config.json", "application/json"),
+          # config tells Watson what you want your document converted to (JSON/HTML/PLAIN TEXT)
+          # This is in a separate file config.json (see below)
+        file: Faraday::UploadIO.new(open(@doc), "application/pdf")
+          # In production, calling open on the @doc url with open-uri allows us
+          # to open the url as if it were a file
+      }
+    else
+      payload = {
+        config: Faraday::UploadIO.new("#{::Rails.root}/lib/WatsonApi/config.json", "application/json"),
+        file: Faraday::UploadIO.new(@doc, "application/pdf")
+      	# This is the file and type to be uploaded. In development, @doc is a
+      	# local file path so it can be sent without "opening" it first
+      }
+    end
+
+    response = conn.post(path, payload)
+    json_response_body = JSON.parse(response.body)
```

Comment:
variable doesn't need to get set if this method is returning `JSON.parse(response.body)`